global_planner: enable default_tolerance with potential field, fix #262

nlimpert opened the pull request:
Previously the default_tolerance parameter was not used leading to
failure when the desired goal was inside an obstacle.
In case the goal is inside an obstacle the global_planner now tries to
find a cell with pot_field_max_cost as its maximum value as close as
possible around the desired goal pose. This is achieved by calculating a
potential field in the area of the desired goal pose.
This area is defined by the default_tolerance parameter.

@mikeferguson this should fix #262.
See a demonstration video at https://fh-aachen.sciebo.de/s/SReCNmR5Larlxt4

vrabaud committed:
[tf2] [WIP] cherry-pick: convert to tf2

Cherry-pick a4a3da9e from #458 & @vrabaud

Original PR branch was aimed at Kinect, thus a straight cherry-pick
does not compile.
This fixes all conflicts detected by Git.

Not Compiling: move_base, global_planner.
Due to changes merged since the original #458.
I will fix those issues in a new commit.

 Conflicts:
	amcl/package.xml
	amcl/src/amcl_node.cpp
	amcl/test/basic_localization.py
	base_local_planner/include/base_local_planner/local_planner_util.h
	base_local_planner/package.xml
	base_local_planner/src/local_planner_util.cpp
	base_local_planner/src/trajectory_planner_ros.cpp
	carrot_planner/package.xml
	clear_costmap_recovery/package.xml
	costmap_2d/CMakeLists.txt
	costmap_2d/include/costmap_2d/costmap_2d_ros.h
	costmap_2d/package.xml
	costmap_2d/src/costmap_2d_ros.cpp
	costmap_2d/test/footprint_tests.cpp
	dwa_local_planner/package.xml
	fake_localization/fake_localization.cpp
	fake_localization/package.xml
	global_planner/package.xml
	move_base/package.xml
	move_base/src/move_base.cpp
	nav_core/package.xml
	navfn/package.xml
	robot_pose_ekf/CMakeLists.txt
	robot_pose_ekf/include/robot_pose_ekf/odom_estimation.h
	robot_pose_ekf/include/robot_pose_ekf/odom_estimation_node.h
	robot_pose_ekf/package.xml
	robot_pose_ekf/src/odom_estimation.cpp
	robot_pose_ekf/src/odom_estimation_node.cpp
	rotate_recovery/package.xml
	rotate_recovery/src/rotate_recovery.cpp

vrabaud committed:
[tf2] cherry-pick: amcl stripSlash

This may have already been forward ported with PR #728
But it was originally done as part of PR #458.

Which is why there was a conflict.
I am cherry-picking for completeness.

get code to work with frames starting with "/"
This is to get test_rosie_multilaser.xml to work

 Conflicts:
	amcl/src/amcl_node.cpp

Commit:
recompute cluster stat when force_publication

When force_publication is True, and no resample has been done the statistic
about the current clusters of particles are not recomputed. So amcl published
an outdated information.

This commit introduces a call to `pf_cluster_stats` when required (i.e no
resample but force_publication).

selvasamuel committed:
Fixed deadlock when changing global planner

Commit:
Fix for #517: create a getRobotPose method on move_base instead of using that on the costmaps

vrabaud committed:
[tf2] cherry pick: Move some packages to tf2

This cherry-picks commit 144cae29

Conflicts:
	base_local_planner/package.xml
	base_local_planner/src/trajectory_planner_ros.cpp
	carrot_planner/package.xml
	global_planner/src/orientation_filter.cpp

 Author:    Vincent Rabaud <vincent.rabaud@gmail.com>
 Date:      Mon Oct 24 22:08:55 2016 +0200

Commit:
Inflation Layer protected members and virtual computeCost [ABI BREAKING]

codebot committed:
Add cost function to prevent unnecessary spinning

Adds a cost function and configuration parameter to penalize changes
in robot heading. This isn't relevant for diff-drive robots, where
heading changes are typically necessary to move towards goals.
However, this is useful for having control of the 'twirling' behavior
of holonomic robots. Setting the twirling_scale parameter to a high
value, say 50, causes the dwa planner to prefer to 'strafe' towards
goals rather than (eventually) aiming the robot at the goal while
driving. For some applications this is much preferred. This parameter
defaults to zero, which maintains the existing behavior unless it is
explicitly set to a non-zero value.

Commit:
amcl: fix compilation with gcc v7

When building amcl with recent enough gcc v7 compilation fails with
the error

src/amcl/map/map_cspace.cpp: In function 'void enqueue(map_t*, unsigned int, unsigned int, unsigned int, unsigned int, std::priority_queue<CellData>&, CachedDistanceMap*, unsigned char*)':
src/amcl/map/map_cspace.cpp:98:34: error: call of overloaded 'abs(unsigned int)' is ambiguous
   unsigned int di = abs(i - src_i);

Use `int abs(int)` flavour of the abs() function.

lengly committed:
fix typo

mikaelarguedas committed:
pcl proagate -lQt5::Widgets flag so we need to find_package Qt5Widgets (#578)

mikaelarguedas committed:
import only PCL common

mikaelarguedas committed:
add missing deps on libpcl

mikaelarguedas committed:
convert packages to format2

mikaelarguedas committed:
alphabetize dependency list

mikaelarguedas committed:
set message_generation build and runtime dependency

mikeferguson committed:
Merge pull request #590 from mikaelarguedas/remove_qt5_dependency

Cleanup PCL imports: fixes compilation on zesty and stretch

mintar committed:
Fix CMakeLists + package.xmls (#548)

* Fix CMakeLists + package.xmls

This fixes compilation errors when rosjava is installed on the system.
It also removes a lot of errors reported by catkin_lint.

Fixes #537 .

* Fix more CMakeLists.txt + package.xmls

* amcl: disambiguate include path

This fixes the following catkin_lint warning:

    amcl: warning: include paths 'include/amcl/map' and 'include' are ambiguous
         * You have used two include paths where one is a parent of the
         * other. Thus the same headers can be included with two different
         * include paths which may confuse users. It is recommended that
         * you keep your include paths consistent.

wjwwood committed:
refactor to not use tf version 1 (#561)

mikeferguson committed:
Merge pull request #596 from ros-planning/lunar_548

Port #548 to lunar

corot committed:
Speedup (~60%) inflation layer update (#525)

* Replace the priority_queue used on inflation layer by a map containing cells to inflate split by distance from the closest obstacle

* Update inflation test to match previous change

mikeferguson committed:
don't update costs if inflation radius is zero

mikeferguson committed:
cleaner logic, fixes #156

mikaelarguedas committed:
rebase fixups

mikeferguson committed:
Merge pull request #600 from ros-planning/lunar_403

recompute cluster stat when force_publication

mikeferguson committed:
Merge pull request #598 from ros-planning/fix_156

cleaner logic, fixes #156

mikeferguson committed:
Merge pull request #591 from mikaelarguedas/package_format2

Convert to package format 2

allenh1 committed:
Fix compiler warning for GCC 8.

mikeferguson committed:
Merge pull request #602 from ros-planning/lunar_534

Inflation Layer protected members and virtual computeCost [ABI BREAKING]

mikeferguson committed:
Merge pull request #601 from ros-planning/lunar_520

#520 for lunar [ABI Breaking]

mikeferguson committed:
Merge pull request #603 from allenh1/fix-gcc-8-warning

Fix compiler warning for GCC 8.

vik748 committed:
Reference Issue #592 Added warning to AMCL when map is published on ... (#604)

hiddewie committed:
Added parameter for allowing inflation in unknown cells (#564)

corot committed:
Add a max_planning_retries parameter to move_base [kinetic] (#539)

* Implements issue #496: add a max_planning_retries parameter as an alternative to planner_patience to limit the failed calls to global planner
* Includes #498 and #524 for kinetic and above

mikeferguson committed:
Merge pull request #607 from ros-planning/lunar_564

Added parameter for allowing inflation in unknown cells (#564)

vrabaud committed:
switch fake_localization to tf2.

mikeferguson committed:
update changelogs

mikeferguson committed:
1.15.0

mikeferguson committed:
remove offending library export (fixes #612)

* remove LIBRARY export that is broken in SDL. public builds were
  working fine before this, trying to quickly fix them.
* remove INCLUDE exports as well, since none of our headers actually
  include SDL headers.

mikeferguson committed:
Merge pull request #615 from ros-planning/fix_l612

remove offending library export (fixes #612)

mikeferguson committed:
update changelogs

mikeferguson committed:
1.15.1

DLu committed:
Closes #625 (#627)

allenh1 committed:
Use occupancy values when saving a map (#613)

DLu committed:
Rebase PRs from Indigo/Kinetic (#637)

* Respect planner_frequency intended behavior (#622)

* Only do a getRobotPose when no start pose is given (#628)

Omit the unnecessary call to getRobotPose when the start pose was
already given, so that move_base can also generate a path in
situations where getRobotPose would fail.

This is actually to work around an issue of getRobotPose randomly
failing.

* Update gradient_path.cpp (#576)

* Update gradient_path.cpp

* Update navfn.cpp

* update to use non deprecated pluginlib macro (#630)

* update to use non deprecated pluginlib macro

* multiline version as well

* Print SDL error on IMG_Load failure in server_map (#631)

aaronhoy committed:
Add myself as a maintainer.

mikeferguson committed:
Merge pull request #649 from aaronhoy/lunar_add_ahoy

Add myself as a maintainer.

DLu committed:
Remove Dead Code [Lunar] (#646)

* Clean up navfn

* Cleanup amcl

DLu committed:
Fixes #313 (Lunar) (#655)

luator committed:
CostmapModel: Make lineCost and pointCost public (#658)

Make the methods `lineCost` and `pointCost` of the CostmapModel class
public so they can be used outside of the class.

Both methods are not changing the instance, so this should not cause any
problems.  To emphasise their constness, add the actual `const` keyword.

SteveMacenski committed:
removing recomputation of cluster stats causing assertion error (#634)

Commit:
fix 'enable' for static_layer with rolling window (#659) (#665)

B0gdar committed:
Update laser_model_type enum on AMCL.cfg

Adding likelihood_field_prob laser model option on AMCL.cfg to be able to control dynamic parameters with this laser sensor model.

DLu committed:
Fixes #206 for kinetic (#663)

* Fixes #206 for kinetic

mikeferguson committed:
update maintainer email

mikeferguson committed:
Merge pull request #670 from DLu/fix206_lunar

Fixes #206 for Lunar

mikeferguson committed:
update build farm links

mikeferguson committed:
fix buildfarm link

stereoboy committed:
Fixed race condition with costmaps in LayeredCostmap::resizeMap()

LayeredCostmap::updateMap() and LayeredCostmap::resizeMap() write to the master grid costmap.
And these two functions can be called by different threads at the same time.
One example of these cases is a race condition between subscriber callback thread
dealing with dynamically-size-changing static_layer and periodical updateMap() calls from Costmap2DROS thread.

Under the situation the master grid costmap is not thread-safe.
LayeredCostmap::updateMap() already used the master grid costmap's lock.

mikeferguson committed:
Merge pull request #673 from ros-planning/email_update_lunar

update maintainer email (lunar)

mikeferguson committed:
Merge pull request #677 from ros-planning/lunar_634

removing recomputation of cluster stats causing assertion error (#634)

mikeferguson committed:
Fix minor typo (#682)

This typo caused some confusion because we were searching for a semicolon in our configuration.

mikeferguson committed:
update changelogs

mikeferguson committed:
1.15.2

mikeferguson committed:
Merge pull request #686 from ros-planning/lunar_675

Fixed race condition with costmaps in LayeredCostmap::resizeMap()

pavloblindnology committed:
Make trajectory scoring scales consistent.

mikeferguson committed:
initialize all costmap variables

mikeferguson committed:
Merge pull request #690 from ros-planning/lunar_609

switch fake_localization to tf2.

mikeferguson committed:
remove unused got_footprint_

mikeferguson committed:
Merge pull request #694 from ros-planning/lunar_691

costmap variable init & cleanup (forward port of #691)

pavloblindnology committed:
Fix trajectory obstacle scoring in dwa_local_planner.

DLu commented:
There's quite a lot going on here, but shouldn't the logic be based on `tolerance` instead of `default_tolerance_`?

romainreignier committed:
Map server wait for a valid time, fix #573 (#700)

When launching the map_server with Gazebo, the current time is picked
before the simulation is started and then has a value of 0.
Later when querying the stamp of the map, a value of has a special
signification on tf transform for example.

mikeferguson committed:
Merge pull request #704 from DLu/fix573_lunar

Map server wait for a valid time fix [lunar]

nlimpert commented:
Of course you are right - Thanks for the notice!

SteveMacenski committed:
adding mutex locks to costmap clearing service

moriarty committed:
[melodic] tf2_buffer_ -> tf2_buffer_ptr_

Change required due to changes in upstream dependencies:
ros/geometry#163: "Maintain & expose tf2 Buffer in shared_ptr for tf"

fixes ros-planning/navigation#717 (for compile errors at least.)

mikeferguson committed:
Merge pull request #719 from ros-planning/lunar_711

adding mutex locks to costmap clearing service

DLu committed:
fix param names of RotateRecovery, closes #706

mikeferguson committed:
Merge pull request #718 from moriarty/tf2-buffer-ptr

[melodic] tf2_buffer_ -> tf2_buffer_ptr_

moriarty committed:
fixes pluginlib deprecated header warnings

moriarty committed:
include <memory> for std::shared_ptr

moriarty committed:
fix test: abs(unsigned int) is ambiguous

Instead, compare values and subtract smaller from larger to find
the dx and dy.

moriarty committed:
[costmap_2d/test] set empty transform to Identity

mikeferguson committed:
Merge pull request #723 from moriarty/melodic-buildfarm-errors

Melodic buildfarm errors

mikeferguson committed:
add const to getLocalPlane, fixes #709

mikeferguson committed:
remove robot_pose_ekf, see #701

mikeferguson committed:
unify combination_method dynamic reconfig, closes #402

mikeferguson committed:
fix param to min_in_place_vel_theta, closes #487

mikeferguson committed:
unify parameter names between base_local_planner and dwa_local_planner

addresses parts of #90

mikeferguson committed:
fix swapped odom1/4 in omni model, fixes #499

mikeferguson committed:
Merge pull request #727 from ros-planning/melodic_668

Update laser_model_type enum on AMCL.cfg (Melodic port of #668)

mikeferguson committed:
switch AMCL to use TF2

cordelliv committed:
Fix for Potential Memory Leak  in AmclNode::reconfigureCB

mikeferguson committed:
Merge pull request #730 from Glowcloud/melodic-devel

Fix for Potential Memory Leak  in AmclNode::reconfigureCB #729

maracuya-robotics committed:
AMCL dynamic reconfigure: Extend parameter range

In Addition to #696, which checks the input arguments, this extends the
allowed parameter range in dynamic reconfigure. The description for
"save_pose_rate" says "-1.0 to disable" but allowed only positive values
previously.

Also increase maximum value for laser_max_beams for high performance
systems.

mganeff committed:
Fixed typos for base_local_planner

mganeff committed:
Fixed typo for test_robot_pose_ekf

mikeferguson committed:
Merge pull request #732 from marting87/small_typo_fixes

Small typo fixes in ftrajectory_planner_ros and robot_pose_ekf

mikeferguson committed:
Merge pull request #725 from ros-planning/first_melodic_release

API/ABI breakers for melodic

mikeferguson committed:
Merge pull request #728 from ros-planning/melodic_tf2_conversion

switch AMCL to use TF2

ipa-fez committed:
Allow specification of free/occupied thresholds for map_saver (#708)

* add occupied threshold command line parameter to map_saver (--occ)
* add free threshold command line parameter to map_saver (--free)

mikeferguson committed:
Merge pull request #734 from ros-planning/melodic_731

AMCL dynamic reconfigure: Extend parameter range (Forward port #731)

mganeff committed:
Minor comment corrections

mikeferguson committed:
Merge pull request #737 from marting87/minor_comment_fixes

Minor comment corrections

mikeferguson committed:
Merge pull request #735 from ros-planning/melodic_708

Allow specification of free/occupied thresholds for map_saver (#708)

mikeferguson commented:
@DLu looks like you started to review this, it's another global_planner one if you have a chance.

DLu commented on global_planner/include/global_planner/planner_core.h:
```diff
@@ -170,6 +172,7 @@ class GlobalPlanner : public nav_core::BaseGlobalPlanner {
          * @brief Store a copy of the current costmap in \a costmap.  Called by makePlan.
          */
         costmap_2d::Costmap2D* costmap_;
+        costmap_2d::Costmap2DROS* costmap_ros_;
```
It doesn't look like you use costmap_ros_ anywhere in this PR.

DLu commented on global_planner/include/global_planner/planner_core.h:
```diff
@@ -205,6 +209,15 @@ class GlobalPlanner : public nav_core::BaseGlobalPlanner {
 
+        unsigned int pot_field_max_cost_;
+
+        std::vector<float> proj_modes;
```
Likewise, proj_modes is not used here either.

DLu commented on global_planner/include/global_planner/planner_core.h:
```diff
@@ -205,6 +209,15 @@ class GlobalPlanner : public nav_core::BaseGlobalPlanner {
+        double new_goal_x;
+        double new_goal_y;
+        double last_official_goal_x;
+        double last_official_goal_y;
```
The standard style in this code is for member variables to end with an underscore.

DLu commented on global_planner/include/global_planner/planner_core.h:
```diff
@@ -205,6 +209,15 @@ class GlobalPlanner : public nav_core::BaseGlobalPlanner {
+        std::vector<float> proj_modes;
+
+        double new_goal_x;
+        double new_goal_y;
```
Why did you make `new_goal_x` and `y` member variables and not just local variables?

DLu commented on global_planner/src/planner_core.cpp:
```diff
@@ -143,6 +144,7 @@ void GlobalPlanner::initialize(std::string name, costmap_2d::Costmap2D* costmap,
         private_nh.param("planner_window_y", planner_window_y_, 0.0);
         private_nh.param("default_tolerance", default_tolerance_, 0.0);
         private_nh.param("publish_scale", publish_scale_, 100);
+        private_nh.param("pot_field_min_", publish_scale_, 100);
```
A) parameter names should not end with an underscore.
B) It probably shouldn't be setting publish_scale, but I'm not sure what its supposed to set either.

DLu commented on global_planner/src/planner_core.cpp:
```diff
@@ -222,6 +227,15 @@ bool GlobalPlanner::makePlan(const geometry_msgs::PoseStamped& start, const geom
+      new_goal_x = goal.pose.position.x;
+      new_goal_y = goal.pose.position.y;
+      ROS_DEBUG("NEW GOAL, setting last goal to: %f %f", new_goal_x, new_goal_y);
+    }
```
If `initialize` is not called, then `last_official_goal_x` will not be initialized, so either, you should
 * Move the initialization to the constructor's initializers (preferred)
 * or Move this if statement to after the initialized check on the next line.

DLu commented on global_planner/src/planner_core.cpp:
```diff
@@ -233,11 +247,12 @@ bool GlobalPlanner::makePlan(const geometry_msgs::PoseStamped& start, const geom
 
     ros::NodeHandle n;
     std::string global_frame = frame_id_;
+    geometry_msgs::PoseStamped goal_ = goal;
```
Likewise, local variables like `goal_` should NOT end in underscores.

DLu commented on global_planner/include/global_planner/planner_core.h:
```diff
@@ -38,6 +38,7 @@
  *         David V. Lu!!
  *********************************************************************/
 #define POT_HIGH 1.0e10        // unassigned cell potential
+#define POT_FIELD_TIMEOUT 1.0  // Timeout of potential field method
```
This should probably be a ROS parameter instead of a `#define`

DLu commented on global_planner/src/planner_core.cpp:
```diff
@@ -247,6 +262,107 @@ bool GlobalPlanner::makePlan(const geometry_msgs::PoseStamped& start, const geom
+        pot_field_target_phi = angles::normalize_angle(atan2(pot_field_target_y, pot_field_target_x));
+
+        float cur_x = std::cos(pot_field_target_phi) * (costmap_->getResolution() * 0.5);
+        float cur_y = std::sin(pot_field_target_phi) * (costmap_->getResolution() * 0.5);
```
Can you explain what this math is doing, please?

DLu reviewed (changes requested):
Thanks for this pull request. The fact that the tolerance was not used to begin with was an oversight on my part. 

There's a couple key stylistic things and extraneous bits of code that make this PR a little tricky. However, my main question is about how exactly you're picking the new goal location. At the very least, it could use some more comments about how that calculation is done. There's also [a nice-ish implementation in `move_base`](https://github.com/ros-planning/navigation/blob/melodic-devel/move_base/src/move_base.cpp#L376) that just searches with increasing distances, which is a little more straightforward.

reinzor committed:
feat(orientation_filter): Added additional orientation filter options (#739)

* feat(orientation_filter): Added additional orientation filter options

Enables plan references with different orientations for omni-base
controllers. The following options are added:

- Backward (backward path traversal, pose points to previous point)
- Leftward (lateral path traversal in the positive y direction)
- Rightward (lateral path traversal in the negative y direction)

* Updated orientation filter option description

* Added window size parameter to orientation filter

Previously, the orientation was calculated using the current and the
next point. However, when the path is somewhat jumpy, this results in
poor orientations. By adding this parameter and altering the orientation
calculation, the calculated orientation can be smoothened along the path
by taking into account a larger window. The orientation of index point i
will be calculated using the positions of i - window_size and i +
window_size.

mikeferguson committed:
Merge pull request #752 from moriarty/vrabaud-kinetic-tf2-math-on-melodic

[tf2] cherry pick: Move some packages to tf2

moriarty committed:
[tf2][Move Base] finish compile error fixes

This finishes the compile errors from cherry-picking #458.
This commit ports the changes from #601 to tf2.

Commit:
Fix: Replaced tf2 setEuler calls with setRPY calls

Introduced by https://github.com/ros-planning/navigation/pull/752

mikeferguson committed:
Merge pull request #762 from ruvu/fix/tf2_set_yaw

Fix: Replaced tf2 setEuler calls with setRPY calls

moriarty committed:
[tf2] setEuler(z,_,_)->setRPY(_,_,z)

moriarty committed:
[whitespace] cleans up some whitepace

moriarty committed:
[tf2][base local planner] puts const back

mikeferguson committed:
Merge pull request #763 from ros-planning/melodic-699

Local Planner Fixes

mikeferguson committed:
Merge pull request #755 from moriarty/vrabuad-kinetic-tf2-on-melodic

tf2 for Melodic

mikeferguson committed:
reorder maintainers slightly

The release announcement script only sees first maintainer of
each package. This way all 3 maintainers get listed, rather than
just David

mikeferguson committed:
remove left over PCL depends in dwa_local_planner

mikeferguson committed:
remove PCL usage from navfn publisher

mikeferguson committed:
Merge pull request #766 from ros-planning/navfn_no_pcl

remove PCL usage from navfn publisher

DLu committed:
Remove PCL from local planners

DLu committed:
Merge remote-tracking branch 'origin/melodic-devel' into remove_pcl

DLu committed:
Remove PCL missing line from navfn

DLu committed:
Remove costmap_2d pcl dep

mikeferguson committed:
Merge pull request #765 from ros-planning/remove_pcl

remove left over PCL depends in dwa_local_planner

mikeferguson committed:
update changelogs

mikeferguson committed:
1.16.0

mikeferguson committed:
make AMCL depend on sensor_msgs

previously, amcl depended on TF, which depended on
sensor_msgs.

mikeferguson committed:
add tf2_geometry_msgs depend to costmap_2d

mikeferguson committed:
Merge pull request #770 from ros-planning/fix_debians

Fix debian builds (closes #769)

mikeferguson committed:
update build farm status link

mikeferguson committed:
update changelogs

mikeferguson committed:
1.16.1

mikeferguson committed:
update amcl to have proper depends

* add geometry_msgs
* add tf2_msgs
* fix alphabetical order

mikeferguson committed:
add explicit sensor_msgs, tf2 depends for base_local_planner

mikeferguson committed:
fix rotate_recovery debian build

* add depend on tf2_geometry_msgs (due to https://github.com/ros/geometry2/issues/275)
* add other hidden depends: angles, geometry_msgs, tf2

mikeferguson committed:
fix depends for dwa_local_planner

* add tf2_geometry_msgs (due to https://github.com/ros/geometry2/issues/275)
* add missing depends on angles, sensor_msgs, tf2

mikeferguson committed:
fix depends of carrot_planner

* declare direct dependency on tf2
* alphabetize the depends in CMake

mikeferguson committed:
update costmap_2d cmake

* explicit dependency on tf2
* remove old PCL disable crap

mikeferguson committed:
Merge pull request #773 from ros-planning/packaging_fixes

packaging fixes

mikeferguson committed:
update changelogs

mikeferguson committed:
1.16.2

mintar committed:
amcl: Set C++ standard 11 if not set

This is required to build the melodic-devel branch of the navigation
stack on kinetic. Melodic sets CMAKE_CXX_STANDARD=14, but kinetic
doesn't set that variable at all.

mikeferguson committed:
Merge pull request #785 from mintar/amcl_c++11

amcl: Add compile option C++11

ipa-fez committed:
remove const from create_nav_plan_astar

mikeferguson committed:
Merge pull request #789 from ipa-fez/fix/astar_const_melodic

Remove const from create_nav_plan_astar

SteveMacenski committed:
fix typo for parameter beam_skip_error_threshold but bandaged for other users in AMCL (#790)

* fix typo but bandage for other users

DLu committed:
Add additional linked libraries (#803)

DLu committed:
Fix for #805 (#813)

DLu committed:
Remove leading slashes from default frame_id parameters

mikeferguson committed:
Merge pull request #831 from ros-planning/feature/remove_slashes

[melodic] Remove leading slashes from default frame_id parameters

mikeferguson committed:
Remove travis

reinzor committed:
feat: AMCL Diagnostics (#807)

Diagnostic task that monitors the estimated standard deviation of the filter.
By: reinzor <reinzor@gmail.com>

DLu committed:
Fix #845 (#846)

DLu committed:
[kinetic] Fix for adjusting plan resolution (#819)

* Fix for adjusting plan resolution

* Simpler math expression

Commit:
AMCL windows build bring up.
* Add HAVE_UNISTD and HAVE_DRAND48 and portable_utils.hpp for better cross compiling.
* Variable length array is not supported in MSVC, conditionally disable it.
* Fix install location for shared lib and executables on Windows.
* Use isfinite for better cross compiling.

Commit:
revert unrelated changes.

Commit:
map_server Windows build bring up
* Fix install location for Windows build. (On Windows build, shared library uses RUNTIME location, but not LIBRARY)
* Use boost::filesystem::path to handle path logic and remove the libgen.h dependency for better cross platform.
* Fix gtest hard-coded and add YAML library dir in CMakeList.txt.

DLu committed:
Fix install (#855)

jspricke committed:
Add missing header

mikeferguson committed:
Merge pull request #857 from jspricke/add_include

Add missing header

IronClimber committed:
remove unused costmap_pub_frequency (#858)

DLu committed:
Set footprint before in place rotation continuation (#829) (#861)

* Make sure to call setFootprint() before an in-place rotation

* Change to const reference

* Remove footprint from findBestPath

IronClimber committed:
Remove unused visualize_potential (#866)

DLu committed:
[kinetic] Fix Bounds Bug (plus test) (#871) (#875)

* fix map bounds bug

* Add test

nlimpert commented:
I'm sorry for the super-late response!
I agree that there are major flaws in this PR and I also agree that the solution implemented in `move_base` is much more elegant.

So I currently tend to re-implement this based on the [solution](https://github.com/ros-planning/navigation/blob/b52e0ea9362131dc81326da471792ab8698c9564/move_base/src/move_base.cpp#L376) in `move_base` that @DLu pointed at.

However seeing this makes me wonder why this would need to be implemented in two different places.
But I guess it's better to have a working `default_tolerance` parameter even though the feature is already available in `move_base` (while it doesn't seem to be usable without `move_base`'s `planService`).

I think this would look completely different than the current PR. Would you like me to create a new PR or simply update this one?

SteveMacenski committed:
clear area in layer for melodic

mikeferguson committed:
Merge pull request #849 from seanyen/amcl_windows_fix

[Windows][melodic] AMCL Windows build bring up.

mikeferguson committed:
Merge pull request #850 from seanyen/map_server_windows_fix

[Windows][melodic] map_server Windows build bring up

sunilsulania committed:
Added publishZeroVelocity() before starting planner (#751)

Edit for Issue #750

DLu committed:
Changed logic for when to resize layered costmap in static layer (#792)

* Changed logic for when to resize layered costmap in static layer

-Now the master layered costmap should no longer get resized when
isSizeLocked returns true

* Fixing format for if loop

corot committed:
Provide different negative values for unknown and out-of-map costs (#833)

corot committed:
Add force_updating and affected_maps parameters to tailor clear costmaps recovey (#838)

behavior

marting87 committed:
Costmap_2d plugin universal parameters and pre-hydro warnings (#738)

* Comment and description clarification

* Renamed resetOldParameters to loadOldParameters

* Upscaled pre-hydro parameter info message to warning and added costmap-name

* Warn user when static_map or map_type is set but not used while plugins are used

* Added function that copies parent parameters inside each layer (makes it possible to set a global inflation_radius)

* use parameter magic

DLu committed:
Fixes #782 (#892)

DLu committed:
Drop Parameter Magic (#893)

hadiTab committed:
Fix typo in amcl_laser model header (#918)

DLu committed:
Cherry pick #914 (#919)

SteveMacenski committed:
Merge branch 'melodic-devel' into layer_clear_area-melodic

mikeferguson committed:
Merge pull request #877 from SteveMacenski/layer_clear_area-melodic

[melodic] moving clearing area method to costmap_layer so other applications can clear other types.

mikeferguson committed:
update changelogs

mikeferguson committed:
1.16.3

Commit:
Fix move_slow_and_clear sending obsolete params to dwa_local_planner
issue.

Commit:
Optimize costmap_2d::updateOrigin

nickswalker committed:
Add frame ID to empty NavFn paths (#964)

* Don't publish empty paths

RViz will complain about not being able to transform the path if it doesn't have a frame ID, so we'll just drop these

Closes #963

* Publish empty paths with a valid frame

* Fix indexing into empty plan for timestamp

nlimpert committed:
amcl: add missing CMake include(CheckSymbolExists)

Starting with CMake 3.15 an explicit include(CheckSymbolExists)
is required to use the check_symbol_exists macro.

mikeferguson committed:
Merge pull request #965 from nlimpert/nlimpert/fix_missing_cmake_include

Add missing CMake include(CheckSymbolExists) for CMake >= 3.15

mikeferguson committed:
Merge pull request #957 from Blindnology/melodic-updateOrigin

Optimize costmap_2d::updateOrigin

mikeferguson committed:
Merge pull request #955 from isjfk/melodic-devel

Fix move_slow_and_clear sending obsolete params to dwa_local_planner

patrickcjh committed:
Set proper limits for the z-weights [amcl]. (#953)

acxz committed:
remove signals dep (#945)

Boost > 1.7 has signals by default

lopsided98 committed:
amcl: include missing CMake functions to fix build (#946)

sumejko92 committed:
parameter for enabling outlineMap added (#926)

seanyen committed:
[Windows][melodic] Navigation (except for map_server and amcl) Windows build bring up (#851)

patrickcjh committed:
Fix some reconfigure parameters not being applied [amcl]. (#952)

nlimpert committed:
commit to be reworked later

nlimpert commented:
Closing this as it will be targeting melodic and it needs a proper rework of the proposed feature.

I'll open a new PR once I'm ready.
